### DISCUSSION TITLE
nested big object ,is slow

### DISCUSSION THREAD

**Discussion by me-v2** [pydantic/pydantic#5173] [Question]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

[response_1678526996187.json](url)

to 100ms.

### Example Code

```Python
class SatTreeSchema(BaseModel):
    id: int
    name: str  # 学科名称
    level: int  # 学科层级
    children: list['SatTreeSchema'] | None = None

    class Config:
        orm_mode = True
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.6
            pydantic compiled: True
                 install path: /Users/hiro/Library/Caches/pypoetry/virtualenvs/argon-QwkX3TcS-py3.11/lib/python3.11/site-packages/pydantic
               python version: 3.11.2 (main, Feb 16 2023, 02:55:59) [Clang 14.0.0 (clang-1400.0.29.202)]
                     platform: macOS-13.2.1-arm64-arm-64bit
     optional deps. installed: ['dotenv', 'typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [X] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [X] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [X] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by dmontagu** [comment]
The link in your comment to what I assume is meant to be the payload ([response_1678526996187.json](https://github.com/pydantic/pydantic/issues/url)) doesn't seem to be correct.

I suspect you'll get a massive performance improvement for things like this when v2 is released, and I'm going to remove the "bug" label from this issue, but I'll leave it open for us to explicitly check for performance changes if/when you can fix the link to the .json payload.

**Comment by samuelcolvin** [comment]
Thanks @dmontagu, you beat me to the punch.

@linpan, pretty impossible for us to help you without the actual data.

Also, in future please use discussions questions for things like this - this is definitely not a bug, even if it's annoying for you.

Converting this to a question.
